backend/dataall/modules/dataset_sharing/services/share_managers/lf_share_manager.py

```diff
@@ -435,29 +431,28 @@ def revoke_external_account_access_on_source_account(self) -> [dict]:
         client = aws_session.client(
             'lakeformation', region_name=self.source_environment.region
         )
-        revoke_entries = []
-        for table in self.revoked_tables:
-            revoke_entries.append(
-                {
-                    'Id': str(uuid.uuid4()),
-                    'Principal': {
-                        'DataLakePrincipalIdentifier': self.target_environment.AwsAccountId
-                    },
-                    'Resource': {
-                        'TableWithColumns': {
-                            'DatabaseName': table.GlueDatabaseName,
-                            'Name': table.GlueTableName,
-                            'ColumnWildcard': {},
-                            'CatalogId': self.source_environment.AwsAccountId,
-                        }
-                    },
-                    'Permissions': ['DESCRIBE', 'SELECT'],
-                    'PermissionsWithGrantOption': ['DESCRIBE', 'SELECT'],
-                }
-            )
-            LakeFormationClient.batch_revoke_permissions(
-                client, self.source_environment.AwsAccountId, revoke_entries
-            )
+        revoke_entries = [
+            {
+                'Id': str(uuid.uuid4()),
+                'Principal': {
+                    'DataLakePrincipalIdentifier': self.target_environment.AwsAccountId
+                },
+                'Resource': {
+                    'TableWithColumns': {
+                        'DatabaseName': db_name,
+                        'Name': table_name,
+                        'ColumnWildcard': {},
+                        'CatalogId': self.source_environment.AwsAccountId,
+                    }
+                },
+                'Permissions': ['DESCRIBE', 'SELECT'],
+                'PermissionsWithGrantOption': ['DESCRIBE', 'SELECT'],
+            }
+        ]
+
+        LakeFormationClient.batch_revoke_permissions(
+            client, self.source_environment.AwsAccountId, revoke_entries
+        )
```

Comment:
With one is enough? it makes sense as there is only one RAM invitation

Reply:
This method is now called once for each table if that particular table is not being shared anywhere on the target account anymore

Before we were checking if the dataset is shared anywhere anymore and if not revoking each table's external target account access which could be problematic (doing dataset-level share checks and revoking on the table-level... explained more in later comment)